FluentAssertions.Core/Types/MemberInfoAssertions.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using FluentAssertions.Common;
+using FluentAssertions.Execution;
+using FluentAssertions.Primitives;
+
+namespace FluentAssertions.Types
+{
+    /// <summary>
+    /// Contains a number of methods to assert that a <see cref="MemberInfo"/> is in the expected state.
+    /// </summary>
+    [DebuggerNonUserCode]
+    public abstract class MemberInfoAssertions<TSubject, TAssertions> : ReferenceTypeAssertions<TSubject, TAssertions>
+        where TSubject : MemberInfo 
+        where TAssertions : MemberInfoAssertions<TSubject, TAssertions>
+    {
+        /// <summary>
+        /// Asserts that the selected member is decorated with the specified <typeparamref name="TAttribute"/>.
+        /// </summary>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion 
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="reasonArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public AndWhichConstraint<MemberInfoAssertions<TSubject, TAssertions>, TAttribute> BeDecoratedWith<TAttribute>(
+            string because = "", params object[] reasonArgs)
+            where TAttribute : Attribute
+        {
+            return BeDecoratedWith<TAttribute>(attr => true, because, reasonArgs);
```

Comment:
Is it possible to change the other implementation of AndWhichConstraint so that isMatchingAttributePredicate defaults to lambda always returning true?